superset/dashboards/api.py

```diff
@@ -1121,30 +1122,38 @@ def cache_dashboard_screenshot(self, pk: int, **kwargs: Any) -> WerkzeugResponse
             "DashboardRestApi.screenshot", pk=dashboard.id, digest=cache_key
         )
 
-        def trigger_celery() -> WerkzeugResponse:
+        screenshot_kwargs = {
+            "username": get_current_user(),
+            "guest_token": (
+                g.user.guest_token
+                if get_current_user() and isinstance(g.user, GuestUser)
+                else None
+            ),
+            "dashboard_id": dashboard.id,
+            "dashboard_url": dashboard_url,
+            "cache_key": cache_key,
+            "force": True,
+            "thumb_size": thumb_size,
+            "window_size": window_size,
+        }
+
+        def generate_screenshot() -> None:
+            cache_dashboard_screenshot(**screenshot_kwargs)
+
+        if is_feature_enabled("DASHBOARD_ASYNC_SCREENSHOT_GENERATION"):
             logger.info("Triggering screenshot ASYNC")
-            cache_dashboard_screenshot.delay(
-                username=get_current_user(),
-                guest_token=(
-                    g.user.guest_token
-                    if get_current_user() and isinstance(g.user, GuestUser)
-                    else None
-                ),
-                dashboard_id=dashboard.id,
-                dashboard_url=dashboard_url,
-                cache_key=cache_key,
-                force=True,
-                thumb_size=thumb_size,
-                window_size=window_size,
-            )
-            return self.response(
-                202,
-                cache_key=cache_key,
-                dashboard_url=dashboard_url,
-                image_url=image_url,
-            )
+            cache_dashboard_screenshot.delay(**screenshot_kwargs)
+        else:
+            logger.info("Triggering screenshot SYNC")
```

Comment:
Why did you use a thread for the non-async version instead of calling `generate_screenshot` directly?

Reply:
The idea here was to let the cache generation function to proceed without blocking the endpoint response. The subsequent call will start to check for the screenshot and will continue trying if a 404 is returned, keeping the same behavior as with Celery. Any particular concern?

Reply:
My concern is that if the screenshot generation is always async then the feature flag name needs to be changed to something like `USE_CELERY_FOR_DASHBOARD_SCREENSHOTS`. Also, if simply spawning a thread is acceptable for asynchronously generating a screenshot, why use Celery?